src/content/docs/en/reference/dev-toolbar-app-reference.mdx

```diff
@@ -9,271 +9,327 @@ The Astro Dev Toolbar App API allows you to create apps that can be used to exte
-          detail: {
-            state: false,
-          },
-        })
-      );
-    }
-  }
+**Signature:** `on<T>(event: string, callback: (data: T) => void) => void`
+
+Registers a callback to be called when the server sends a message with the specified event.
+
+```ts title="src/my-app.js"
+init(canvas, app, server) {
+  server.on('server-message', (data) => {
+    console.log(data.message);
+  });
+}
 ```
 
-#### `state: boolean`
+### On the server
 
-Indicates whether or not the app should be enabled. When `true`, the app will be enabled. Conversely, when `false`, the app will be disabled. If this property is not specified, `true` will be assumed.
+In your integration's `astro:server:setup` hook, you can use the `toolbar` object to send and receive messages from your apps.
 
-## Client-Server Communication
+#### `send()`
 
-Using [Vite's methods for client-server communication](https://vitejs.dev/guide/api-plugin.html#client-server-communication), Dev Toolbar Apps can communicate with the server.
+**Signature:** `send<T>(event: string, data: T) => void`
 
-In addition to being able to send and receive custom messages, the Dev Toolbar also sends the following messages, where `APP_ID` is the [app's ID](#id-string):
+Sends data to the client.
 
-### `astro-dev-toolbar:APP_ID:initialized`
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.send('server-message', { message: 'Hello!' });
+},
+```
 
-This message is sent when the app is initialized. The data for this message is empty.
+#### `on()`
 
-```ts title="integration.ts" {4-6}
-{
-  // ...
-  "astro:server:setup": ({ server }) => {
-    server.ws.on("astro-dev-toolbar:super-app:initialized", () => {
-      console.log("My app was initialized!");
-    });
-  },
-  // ...
-}
+**Signature:** `on<T>(event: string, callback: (data: T) => void) => void`
+
+Registers a callback to be called when the client sends a message with the specified event.
+
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.on('my-app:my-message', (data) => {
+    console.log(data.message);
+  });
+},
 ```
 
-### `astro-dev-toolbar:APP_ID:toggled`
+#### `onInitialized`
 
-This message is sent when the user clicks on the app icon in the UI. The data for this message is a boolean indicating whether the app is enabled or not.
+**Signature:** `onInitialized(appId: string, callback: () => void) => void`
 
-```ts title="integration.ts" {4-6}
-{
-  // ...
-  "astro:server:setup": ({ server }) => {
-    server.ws.on("astro-dev-toolbar:super-app:toggled", (data) => {
-      console.log(`My app is now ${data.state ? "enabled" : "disabled"}!`);
-    });
-  },
-  // ...
-}
+Registers a callback to be called when the app is initialized.
+
+```ts title="my-integration.js"
+'astro:server:setup': ({ toolbar }) => {
+  toolbar.onInitialized('my-app', () => {
+    console.log('The app is now initialized!');
+  });
+},
 ```
 
 :::note
```

Comment:
Quick note that in the current version of docs, this "note" came after both `astro-dev-toolbar:APP_ID:toggled` and  `astro-dev-toolbar:APP_ID:initialized` as if it applied to both.

In this version, this note is *between* the two. So just double check whether this applies only to `onInitialized` or also to `onToggled`!

Reply:
This is okay! The note is only about how you should use the app's event vs Vite, it doesn't matter for other events.